### PR TITLE
Implement storage for IncomeList and SpendingList

### LINKED ISSUE #21: As a user, I want to store my listings, so that I save my progress

### PR DESCRIPTION
**Pull request by TPH777**
Income and Spending will extend a new class Type, as they share similar attributes and methods.

During initialisation of storage in main, Storage will initialise incomes, spendings and password to restore data. Both `IncomeListStorage` and `SpendingListStorage` will serialise when load (at the start) and deserialise when save (at the end).
-It is not implemented as one class due to the complexity of the data type, for example, `IncomeList extends ArrayList<Income>` and `Income extends Type`. This means that 2 generics have to be used.
  * Unless `IncomeList` and `SpendingList` is `extends ArrayList<Type>`, but this may cause bugs like accidentally adding income to `SpendingList` (Can be discussed).

`LoginStorage` will take the hashed password at the password.txt file if it exist, else it will call `createNewUser` and create a new file for the newly set hashed password.

closes #21

### CODE REVIEW

File: src/main/java/seedu/storage/LoginStorage.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.io.IOException;
+import java.util.Scanner;
+
+public class LoginStorage {
+    private static final String PASSWORD_FILE_PATH = "./password.txt";
+
+    public static int load(Ui ui) {
+        try {
+            File f = new File(PASSWORD_FILE_PATH);
+            boolean isFileCreated = f.exists();
```
Comment (NigelYeoTW) [inline src/main/java/seedu/storage/LoginStorage.java:16]:
```suggestion
            boolean isFileCreated = new File(PASSWORD_FILE_PATH).createNewFile();
```

File: src/main/java/seedu/storage/Storage.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import seedu.classes.Ui;
+import seedu.type.IncomeList;
+import seedu.type.SpendingList;
+
+public class Storage {
+    private static IncomeList incomes;
+    private static SpendingList spendings;
+    private static int password;
+
+    public Storage(Ui ui) {
```
Comment (NigelYeoTW) [inline src/main/java/seedu/storage/Storage.java:12]: Maybe could make the IncomeList, SpendingList and LoginStorage methods package private instead so only the storage package classes can access them

File: src/main/java/seedu/storage/LoginStorage.java (hunk 1 above)
Comment (TPH777) [reply src/main/java/seedu/storage/LoginStorage.java:16]: I find this to be not as readable, as f is still used later on to initialise Scanner

**Comment by NigelYeoTW** [comment]
lgtm

**Pull request merged**